main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -17,18 +17,323 @@ package ca.uwaterloo.flix.language.ast
 
 sealed trait TokenKind
 
+// NOTE: Tokens are named for 'what they are' rather than 'what they represent'.
+// So '::' is not named 'Cons' but instead 'ColonColon' as the lexer should be oblivious to the concept of cons
+
+// NOTE: Builtin type keywords like 'Float32' are lexed into 'Float32Keyword' whereas Float32 *literals* are lexed to 'Float32'
+// This aligns naming of literals with other types such as 'String' and 'Bool'
+
 object TokenKind {
+  case object Ampersand extends TokenKind
+
+  case object AngledEqual extends TokenKind
+
+  case object AngledEqualEqual extends TokenKind
+
+  case object AngledPlus extends TokenKind
+
+  case object Annotation extends TokenKind
+
+  case object Arrow extends TokenKind
+
+  case object At extends TokenKind
+
+  case object BackArrow extends TokenKind
+
+  case object Backslash extends TokenKind
+
+  case object Bang extends TokenKind
+
+  case object Bar extends TokenKind
+
+  case object BigDecimal extends TokenKind
+
+  case object BigInt extends TokenKind
+
+  case object BuiltIn extends TokenKind
+
+  case object Caret extends TokenKind
+
+  case object Char extends TokenKind
+
+  case object Colon extends TokenKind
+
+  case object ColonColon extends TokenKind
+
+  case object ColonEqual extends TokenKind
+
+  case object Comma extends TokenKind
+
+  case object CommentBlock extends TokenKind
+
+  case object CommentLine extends TokenKind
+
+  case object Dollar extends TokenKind
+
+  case object Dot extends TokenKind
+
+  case object DotDot extends TokenKind
```

Comment:
Is two dots used anywhere?

Reply:
No I think I found some leftover from a range notation somewhere. I can't relocate it though!
I removed the tokenkind